### DISCUSSION TITLE
github-releases datasource does not strip "v" prefix

### DISCUSSION THREAD

**Discussion by neeme-praks-sympower** [renovatebot/renovate#21474] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

https://github.com/renovatebot/renovate/blob/1e6c757d01457a57d0ed446e3d4553aeae51a8cb/lib/modules/datasource/github-releases/index.ts#L56-L64

The second point in this list (sanitize) does not seem to be implemented: I just got a PR about [ProtoBuf release](https://github.com/protocolbuffers/protobuf/releases) 22.3 and the version string is `v22.3` -- I would expect it to be `22.3`. It is possible that this has never worked (as I just recently set up Renovate bot on my project and this is the first PR for updating ProtoBuf).

My bug report was closed and I was directed to open this as discussion topic.

**Comment by viceice** [comment]
https://docs.renovatebot.com/configuration-options/#extractversion

**Reply by neeme-praks-sympower** [reply]
Thanks! As a noob, I didn't make the connection from https://docs.renovatebot.com/modules/datasource/github-releases/ to https://docs.renovatebot.com/configuration-options/

**Comment by lmagyar** [comment]
Could you please help? I've spent 2 days figuring out how github-releases works, none of my configurations work.

Dockerfile:
```
ARG PORTAINER_VERSION="2.17.1"
```

New github portainer repo release name: "Release 2.18.1"
New github portainer repo tag: "2.18.1"
See: https://github.com/portainer/portainer/releases

Renovate config:
```
  "regexManagers": [
    {
      "fileMatch": ["/Dockerfile$"],
      "matchStrings": [
        "ARG PORTAINER_VERSION=[\"']?(?<currentValue>.+?)[\"']?\\s+"
      ],
      "datasourceTemplate": "github-releases",
      "depNameTemplate": "portainer/portainer"
    }
  ]
```

It has never detected the new version.

I've tried:
Adding `"extractVersionTemplate": "^Release (?<version>.*)$"` to `regexManagers`, doesn't work.
Adding
```
  "packageRules": [
    {
      "matchDatasources": ["github-releases"],
      "matchDepNames": ["portainer/portainer"],
      "extractVersion": "^Release.(?<version>.*)$"
    }
  ]
```
Or adding
```
  "packageRules": [
    {
      "matchDatasources": ["github-releases"],
      "matchDepNames": ["portainer/portainer"],
      "versioning": "regex:^Release.(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$"
    }
  ]
```
Still not workig.

I've tried adding a "v" to the dockerfile:
```
ARG PORTAINER_VERSION="v2.17.1"
```
Not working.

And nearly all the possible combination of the above, still nothing. It always parses the dockerfile, finds `"currentValue": "2.17.1",`, but never detects the new release. Never.

**Reply by rarkins** [reply]
Please start a separate discussion, and create a reproduction repo so we can take a look

**Reply by lmagyar** [reply]
Thanks, done: https://github.com/renovatebot/renovate/discussions/21617
